src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -167,6 +170,9 @@ private void CancellationCallback()
 
                     // Stop the FS event message pump
                     Interop.RunLoop.CFRunLoopStop(_watcherRunLoop);
+
+                    // Unlock and close the handle on the root directory
+                    _rootHandle.Dispose();
```

Comment:
What happens if processing ends because of an error rather than because of cancellation?